File: android/app/src/main/java/com/kgeneye/eye/qc/QCThresholds.kt
package com.kgeneye.eye.qc

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class Orientation {
    @SerialName("portrait")
    PORTRAIT,

    @SerialName("landscape")
    LANDSCAPE,

    @SerialName("any")
    ANY,
}

@Serializable
data class QCThresholds(
    val minDurationMs: Int = 5_000,
    val maxDurationMs: Int = 600_000,
    val requiredOrientation: Orientation = Orientation.LANDSCAPE,
    val minHandPresenceRate: Double = 0.6,
    val maxFacePresenceRate: Double = 0.15,
    val minReadinessScore: Double = 65.0,
    val warnReadinessScore: Double = 85.0,
    val minStabilityScore: Double = 40.0,
    val minBrightnessScore: Double = 35.0,
    val minBlurScore: Double = 40.0,
)

File: ios/QCEngine/Sources/QCEngine/QCThresholds.swift
public enum Orientation: String, Codable, Equatable {
    case portrait
    case landscape
    case any
}

public struct QCThresholds: Equatable {
    public let minDurationMs: Int
    public let maxDurationMs: Int
    public let requiredOrientation: Orientation
    public let minHandPresenceRate: Double
    public let maxFacePresenceRate: Double
    public let minReadinessScore: Double
    public let warnReadinessScore: Double
    public let minStabilityScore: Double
    public let minBrightnessScore: Double
    public let minBlurScore: Double

    public init(
        minDurationMs: Int = 5_000,
        maxDurationMs: Int = 600_000,
        requiredOrientation: Orientation = .landscape,
        minHandPresenceRate: Double = 0.6,
        maxFacePresenceRate: Double = 0.15,
        minReadinessScore: Double = 65,
        warnReadinessScore: Double = 85,
        minStabilityScore: Double = 40,
        minBrightnessScore: Double = 35,
        minBlurScore: Double = 40
    ) {
        self.minDurationMs = minDurationMs
        self.maxDurationMs = maxDurationMs
        self.requiredOrientation = requiredOrientation
        self.minHandPresenceRate = minHandPresenceRate
        self.maxFacePresenceRate = maxFacePresenceRate
        self.minReadinessScore = minReadinessScore
        self.warnReadinessScore = warnReadinessScore
        self.minStabilityScore = minStabilityScore
        self.minBrightnessScore = minBrightnessScore
        self.minBlurScore = minBlurScore
    }

    public static let `default` = QCThresholds()
}
